### PR TITLE
Create transport

### PR DESCRIPTION
**Pull request by rudigiesler**
Create the USSD transport according to the following spec: [vas2nets_ussd_userguide_1.2.pdf](https://github.com/praekelt/vumi-vas2nets/files/115820/vas2nets_ussd_userguide_1.2.pdf) . Transport will inherit from the vumi HttpRpcTransport base transport.

### CODE REVIEW

**Comment by rudigiesler** [comment]
Relies on https://github.com/praekelt/vumi-vas2nets/pull/1 . Ready for review once https://github.com/praekelt/vumi-vas2nets/pull/1 has landed.

File: vxvas2nets/vas2nets_ussd.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+            )
+            yield self.add_status(
+                component='request', status='down',
+                type='invalid_inbound_fields',
+                message='Invalid inbound fields', details=errors)
+            return
+
+        yield self.add_status(
+            component='request', status='ok', type='request_parsed',
+            message='Request parsed',)
```
Comment (hodgestar) [inline vxvas2nets/vas2nets_ussd.py:97]: I'm wondering about the interaction between this status ok and the slow response down time statuses. If slow responses are coming in, that's mean the status would flip-flop between ok and down or degraded on every message. Perhaps we should leave returning the status to ok for when the response comes back?

File: vxvas2nets/vas2nets_ussd.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+            reasons=[
+                'Response took longer than %fs' % (
+                    self.response_time_degraded,)
+            ],
+            details={
+                'response_time': time,
+            })
+
+    def on_good_response_time(self, message_id, time):
+        request = self.get_request(message_id)
```
Comment (hodgestar) [inline vxvas2nets/vas2nets_ussd.py:198]: Do we have to worry about `request` possibly being `None` here?

File: vxvas2nets/vas2nets_ussd.py (hunk 1 above)
Comment (hodgestar) [inline vxvas2nets/vas2nets_ussd.py:97]: Oh, these are for different components. Woot. :)

File: vxvas2nets/tests/test_vas2nets_ussd.py [hunk 3]
```diff
@@ -0,0 +1,422 @@
...
+            'msisdn': '+123',
+        })
+        self.assertEqual(response.code, 200)
+
+        [ack] = yield self.tx_helper.wait_for_dispatched_events(1)
+        self.assert_ack(ack, reply)
+
+    @inlineCallbacks
+    def test_nack_insufficient_message_fields(self):
+        '''When there are missong fields in a reply message, then we should
```
Comment (hodgestar) [inline vxvas2nets/tests/test_vas2nets_ussd.py]: s/missong/missing/

File: vxvas2nets/vas2nets_ussd.py (hunk 2 above)
Comment (rudigiesler) [inline vxvas2nets/vas2nets_ussd.py:198]: This function is called by `finish_request` inside of its `if request:` block (in the base HttpRpc transport), before the `remove_request` call, so we are guaranteed to have a request at this point.

**Comment by KaitCrawford** [comment]
:+1: I'm happy :smiley:

File: vxvas2nets/vas2nets_ussd.py (hunk 2 above)
Comment (hodgestar) [inline vxvas2nets/vas2nets_ussd.py:198]: Cool.

**Comment by hodgestar** [comment]
:+1:

**Pull request merged**